R/process-dir.R

```diff
@@ -0,0 +1,260 @@
+#' @title
+#' Find a layout file given plate filepath
+#'
+#' @import fs
+#' @importFrom stringr str_split
+#'
+#' @keywords internal
+#'
+find_layout_file <- function(plate_filepath, layout_filepath = NULL) {
+  if (!is.null(layout_filepath)) {
+    stopifnot(fs::file_exists(layout_filepath))
+    return(layout_filepath)
+  }
+
+  stopifnot(fs::is_absolute_path(plate_filepath))
+
+  file_dir <- fs::path_dir(plate_filepath)
+  filename <- fs::path_file(plate_filepath)
+  filename_splitted <- stringr::str_split(filename, "\\.")
+  filename_basename <- filename_splitted[[1]][1]
+
+  supported_layout_exts <- c("xlsx", "csv")
+  layout_file_glob <- paste0(
+    filename_basename, "_layout",
+    "\\.(", paste(supported_layout_exts, collapse = "|"), ")$"
+  )
+  possible_files <- list.files(file_dir, pattern = layout_file_glob)
+  if (length(possible_files) == 0) {
+    stop(
+      paste0("Layout file for a file ", plate_filepath, " could not be found.")
+    )
+  }
+
+  possible_layout_filename <- possible_files[1]
+  possible_layout_path <- fs::path_join(c(file_dir, possible_layout_filename))
+  if (fs::file_exists(possible_layout_path)) {
+    return(possible_layout_path)
+  } else {
+    stop(
+      paste0("Layout file for a file ", plate_filepath, " could not be found.")
+    )
+  }
+}
+
+#' @title
+#' Identify if a file is a MBA data file
+#'
+#' @import fs
+#' @importFrom stringr str_split
+#'
+#' @keywords internal
+#'
+is_mba_data_file <- function(filepath) {
```

Comment:
I'm trying to run this on data from One drive and i get `No files found in the input directory`. Its because files there are named similar to IG7738~1.CSV 
<img width="596" alt="image" src="https://github.com/user-attachments/assets/fc6b4fb6-0257-4315-b1d2-0c093188c4be" />

for this data layout is called: layout_S4_P5.xlsx

I think patterns set here are to restrective and will result in high ratio of no-match cases